File: classifier/data/raw/human/samples/project_euler_zig/problem_017.zig
// Project Euler 017: letters used when writing one to one thousand in English.
const std = @import("std");

fn lettersUnder100(n: u32) u32 {
    const ones = [_]u32{ 0, 3, 3, 5, 4, 4, 3, 5, 5, 4 };
    const teens = [_]u32{ 3, 6, 6, 8, 8, 7, 7, 9, 8, 8 };
    const tens = [_]u32{ 0, 0, 6, 6, 5, 5, 5, 7, 6, 6 };
    if (n < 10) return ones[n];
    if (n < 20) return teens[n - 10];
    return tens[n / 10] + ones[n % 10];
}

fn letters(n: u32) u32 {
    if (n == 1000) return 11; // one thousand
    if (n >= 100) return lettersUnder100(n / 100) + 7 + if (n % 100 == 0) 0 else 3 + lettersUnder100(n % 100);
    return lettersUnder100(n);
}

pub fn main() !void {
    var total: u32 = 0;
    for (1..1001) |n| total += letters(@intCast(n));
    var buffer: [32]u8 = undefined;
    const text = try std.fmt.bufPrint(&buffer, "{d}\n", .{total});
    _ = std.c.write(1, text.ptr, text.len);
}

File: classifier/data/raw/ai/samples/project_euler_java/Problem017.java
// Project Euler Problem 17: letters used when spelling the numbers one through one thousand.
public class Problem017 {
    private static final String[] ONES = {"", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
    private static final String[] TEENS = {"ten", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"};
    private static final String[] TENS = {"", "", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"};

    public static void main(String[] args) {
        int total = 0;
        for (int number = 1; number <= 1000; number++) total += words(number).length();
        System.out.println(total);
    }

    private static String words(int number) {
        if (number == 1000) return "onethousand";
        if (number >= 100) {
            String result = ONES[number / 100] + "hundred";
            return number % 100 == 0 ? result : result + "and" + words(number % 100);
        }
        if (number >= 20) return TENS[number / 10] + ONES[number % 10];
        if (number >= 10) return TEENS[number - 10];
        return ONES[number];
    }
}
